### PR TITLE
Render psd sprite in previews

### PR DESCRIPTION
**Pull request by bigfoodK**
image

### CODE REVIEW

**Comment by namse** [comment]
Re-request review after update please

File: luda-editor/new-client/Cargo.toml [hunk 1]
```diff
@@ -14,3 +14,6 @@ serde = { version = "1.0.203", features = ["derive"] }
 luda-rpc = { path = "../new-server/rpc" }
 dashmap = "6.0.1"
 randum = { path = "../new-server/randum" }
+serde_json = "1.0.125"
```
[the remark this answers is not in the document]
Comment (bigfoodK) [reply luda-editor/new-client/Cargo.toml:17]: on `take_google_gsi_jwt()`

https://github.com/NamseEnt/namseent/blob/018cbb258960cd8556b22d4a33d8b95aaf114c03/luda-editor/new-client/src/lib.rs#L169-L169

**Comment by bigfoodK** [comment]
image

File: luda-editor/new-client/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (namse) [reply luda-editor/new-client/Cargo.toml:17]: i think we removed json export from namui. hmm

File: luda-editor/new-client/src/episode_editor/scene_list.rs [hunk 2]
```diff
@@ -53,9 +63,15 @@ impl Component for SceneListCell<'_> {
...
-        //     sprite.sprite_id
-        // }
-        todo!()
+        ctx.compose(|ctx| {
+            ctx.translate(Xy::single(4.px()))
+                .add(typography::body::left_top(index.to_string(), Color::WHITE));
+        });
+
+        for scene_sprite in &scene.scene_sprites {
+            render_psd_sprite(ctx, scene_sprite, wh);
```
Comment (namse) [inline luda-editor/new-client/src/episode_editor/scene_list.rs:72]: is it ok not to use ctx compose with key for this?

File: luda-editor/psd-sprite-render/src/render_psd_sprite.rs [hunk 3]
```diff
@@ -2,26 +2,52 @@ use crate::blender::photoshop_blend_mode_into_blender;
...
 
-        // TODO: Commented because of compile errors.
-
-        ctx.effect("create image filter", || {
-            match create_image_filter(scene_sprite, self, todo!()) {
-                Ok(image_filter) => set_image_filter.set(image_filter),
-                Err(err) => todo!(),
-            }
-        });
+        if let ImageFilterCreateState::Unset = image_filter_create_state.as_ref() {
```
Comment (namse) [inline luda-editor/psd-sprite-render/src/render_psd_sprite.rs:27]: Not good idea. use effect.

File: luda-editor/new-client/src/render_psd_sprite.rs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    Loading,
+    Loaded {
+        psd_sprite: Arc<PsdSprite>,
+        loaded_images: Arc<SpriteLoadedImages>,
+    },
+    #[allow(unused)]
+    Error(Box<dyn Error + Send + Sync>),
+}
+
+struct PsdSpriteStorage {
```
Comment (namse) [inline luda-editor/new-client/src/render_psd_sprite.rs:62]: where do you use this storage??

**Review by namse — CHANGES_REQUESTED** [review]

File: luda-editor/new-client/src/episode_editor/scene_list.rs (hunk 2 above)
Comment (bigfoodK) [reply luda-editor/new-client/src/episode_editor/scene_list.rs:72]: Seems needed. I will fix in this pr

File: luda-editor/new-client/src/render_psd_sprite.rs (hunk 4 above)
Comment (bigfoodK) [reply luda-editor/new-client/src/render_psd_sprite.rs:62]: https://github.com/NamseEnt/namseent/pull/923/files#diff-8307c749c165fe3cef874cfec9499c563dd5bc4a669b4fb42d81d0aeeaf94d61R11-R13

Here

File: luda-editor/psd-sprite-render/src/render_psd_sprite.rs (hunk 3 above)
Comment (bigfoodK) [reply luda-editor/psd-sprite-render/src/render_psd_sprite.rs:27]: Yes, I will fix it in this pr

File: luda-editor/new-client/src/render_psd_sprite.rs (hunk 4 above)
Comment (namse) [reply luda-editor/new-client/src/render_psd_sprite.rs:62]: then the name `PSD_SPRITE_LOAD_STATE` must be strange. it should be storage. I thought `PSD_SPRITE_LOAD_STATE` is the kind of enum value.

**Review by namse — APPROVED** [review]
LGTM

**Pull request merged**